<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NV200Widget</class>
 <widget class="QWidget" name="NV200Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1376</width>
    <height>994</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="leftMargin">
    <number>9</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <widget class="QComboBox" name="devicesComboBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="searchDevicesButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Search Devices ...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="connectButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>12</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_10">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QScrollArea" name="scrollArea">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>10</height>
            </size>
           </property>
           <property name="frameShape">
            <enum>QFrame::Shape::NoFrame</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>247</width>
              <height>582</height>
             </rect>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>30</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QTabWidget" name="tabWidget">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="tabPosition">
                <enum>QTabWidget::TabPosition::West</enum>
               </property>
               <property name="currentIndex">
                <number>0</number>
               </property>
               <widget class="QWidget" name="easyModeTab">
                <attribute name="title">
                 <string>Easy Mode</string>
                </attribute>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <item>
                  <widget class="QGroupBox" name="easyModeGroupBox">
                   <property name="enabled">
                    <bool>false</bool>
                   </property>
                   <property name="title">
                    <string>Easy Mode</string>
                   </property>
                   <layout class="QVBoxLayout" name="verticalLayout">
                    <item>
                     <widget class="QCheckBox" name="closedLoopCheckBox">
                      <property name="text">
                       <string>Open Loop</string>
                      </property>
                      <property name="toggleSwitch" stdset="0">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <spacer name="verticalSpacer_3">
                      <property name="orientation">
                       <enum>Qt::Orientation::Vertical</enum>
                      </property>
                      <property name="sizeType">
                       <enum>QSizePolicy::Policy::Fixed</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>0</width>
                        <height>10</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                    <item>
                     <widget class="QLabel" name="targetPositionsLabel">
                      <property name="text">
                       <string>Target Positions:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <layout class="QGridLayout" name="gridLayout">
                      <property name="topMargin">
                       <number>0</number>
                      </property>
                      <item row="1" column="0">
                       <widget class="QDoubleSpinBox" name="targetPosSpinBox_2">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="alignment">
                         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                        </property>
                        <property name="decimals">
                         <number>3</number>
                        </property>
                        <property name="maximum">
                         <double>1000.000000000000000</double>
                        </property>
                       </widget>
                      </item>
                      <item row="1" column="1">
                       <widget class="QPushButton" name="moveButton_2">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="text">
                         <string/>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="0">
                       <widget class="QDoubleSpinBox" name="targetPosSpinBox">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="alignment">
                         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                        </property>
                        <property name="prefix">
                         <string/>
                        </property>
                        <property name="suffix">
                         <string/>
                        </property>
                        <property name="decimals">
                         <number>3</number>
                        </property>
                        <property name="maximum">
                         <double>1000.000000000000000</double>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="1">
                       <widget class="QPushButton" name="moveButton">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="text">
                         <string/>
                        </property>
                       </widget>
                      </item>
                      <item row="2" column="0">
                       <widget class="QLabel" name="rangeLabel">
                        <property name="text">
                         <string/>
                        </property>
                        <property name="alignment">
                         <set>Qt::AlignmentFlag::AlignCenter</set>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <spacer name="verticalSpacer">
                   <property name="orientation">
                    <enum>Qt::Orientation::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>740</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
               <widget class="QWidget" name="settingsTab">
                <attribute name="title">
                 <string>Settings</string>
                </attribute>
                <layout class="QVBoxLayout" name="verticalLayout_8">
                 <item>
                  <widget class="QPushButton" name="applyButton">
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Apply Parameters&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Send the currently edited parameters to the device to update its configuration.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Apply Parameters</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="retrieveButton">
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Retrieve Parameters&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Read the current parameters from the device and update the local view.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Retrieve Parameters</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="restoreButton">
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Restore Default Parameters&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Reset the device parameters to their previous values.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Restore Default Parameters</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="verticalSpacer_2">
                   <property name="orientation">
                    <enum>Qt::Orientation::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>654</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
               <widget class="QWidget" name="waveformTab">
                <attribute name="title">
                 <string>Waveform</string>
                </attribute>
                <layout class="QFormLayout" name="formLayout">
                 <item row="0" column="0">
                  <widget class="QLabel" name="freqLabel">
                   <property name="text">
                    <string>Freq.</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QDoubleSpinBox" name="freqSpinBox">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="suffix">
                    <string> Hz</string>
                   </property>
                   <property name="minimum">
                    <double>0.010000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>100.000000000000000</double>
                   </property>
                   <property name="value">
                    <double>20.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="phaseLabel">
                   <property name="text">
                    <string>Phase Shift</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QDoubleSpinBox" name="phaseShiftSpinBox">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="suffix">
                    <string> °</string>
                   </property>
                   <property name="decimals">
                    <number>3</number>
                   </property>
                   <property name="minimum">
                    <double>0.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>360.000000000000000</double>
                   </property>
                   <property name="value">
                    <double>0.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="highLabel">
                   <property name="text">
                    <string>High Level</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QDoubleSpinBox" name="highLevelSpinBox">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="decimals">
                    <number>3</number>
                   </property>
                   <property name="maximum">
                    <double>1000.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="lowLabel">
                   <property name="text">
                    <string>Low Level</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QDoubleSpinBox" name="lowLevelSpinBox">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="decimals">
                    <number>3</number>
                   </property>
                   <property name="maximum">
                    <double>1000.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="0" colspan="2">
                  <widget class="QPushButton" name="uploadButton">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>Upload</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <spacer name="verticalSpacer_4">
                   <property name="orientation">
                    <enum>Qt::Orientation::Vertical</enum>
                   </property>
                   <property name="sizeType">
                    <enum>QSizePolicy::Policy::Fixed</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>0</width>
                     <height>10</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item row="6" column="0">
                  <widget class="QLabel" name="cyclesLabel">
                   <property name="text">
                    <string>Cycles</string>
                   </property>
                  </widget>
                 </item>
                 <item row="6" column="1">
                  <widget class="QSpinBox" name="cyclesSpinBox">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximum">
                    <number>65535</number>
                   </property>
                   <property name="value">
                    <number>1</number>
                   </property>
                  </widget>
                 </item>
                 <item row="7" column="0" colspan="2">
                  <layout class="QHBoxLayout" name="horizontalLayout_6">
                   <item>
                    <widget class="QPushButton" name="startWaveformButton">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="text">
                      <string>Start</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QPushButton" name="stopWaveformButton">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="text">
                      <string>Stop</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="consoleButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Console</string>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QStackedWidget" name="stackedWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="MplWidget" name="mplCanvasWidget"/>
         <widget class="Nv200ControllerWidget" name="controllerStructureWidget"/>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="consoleWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_11">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="consoleLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Command Line Interface</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Console" name="console">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">QTextEdit { background: black; }</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="TimedProgressBar" name="moveProgressBar">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>3</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">QProgressBar { background: transparent;}</string>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="textVisible">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MplWidget</class>
   <extends>QWidget</extends>
   <header location="global">pysoworks.mplcanvas</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TimedProgressBar</class>
   <extends>QProgressBar</extends>
   <header location="global">pysoworks.timed_progress_bar</header>
  </customwidget>
  <customwidget>
   <class>Console</class>
   <extends>QTextEdit</extends>
   <header location="global">pysoworks.consolewidget</header>
  </customwidget>
  <customwidget>
   <class>Nv200ControllerWidget</class>
   <extends>QWidget</extends>
   <header location="global">pysoworks.nv200_controller_widget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
